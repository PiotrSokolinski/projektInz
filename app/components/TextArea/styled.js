import styled from 'styled-components'
import { Colors } from 'themes'

export const TextArea = styled.textarea`
  resize: none;
  outline: none;
  width: 100%;
  border: 1px solid ${Colors.silverChalice};
  border-radius: 5px;
  height: 100px;
  padding: 10px 15px;
  font-size: 15px;
  line-height: 140%;
  letter-spacing: 0.02em;
  box-sizing: border-box;
`
