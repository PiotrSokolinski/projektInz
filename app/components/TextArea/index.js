/**
 *
 * TextArea
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { Field, Control, Help } from 'react-bulma-components/lib/components/form'

import Label from 'components/Label'

import * as Styled from './styled'

const TextArea = ({ error, id, label, textAreaProps }) => (
  <Field id={id}>
    <Label error={error} label={label} />
    <Control>
      <Styled.TextArea {...textAreaProps} />
    </Control>
    {error && <Help color="danger">{error}</Help>}
  </Field>
)

TextArea.propTypes = {
  error: PropTypes.string,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  label: PropTypes.string,
  textAreaProps: PropTypes.object,
}

export default TextArea
