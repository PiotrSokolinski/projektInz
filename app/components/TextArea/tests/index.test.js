/**
 *
 * Tests for TextArea
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'

import TextArea from '../index'

describe('<TextArea />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<TextArea label="Label" />)
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<TextArea label="Label" />)
    expect(firstChild).toMatchSnapshot()
  })
})
