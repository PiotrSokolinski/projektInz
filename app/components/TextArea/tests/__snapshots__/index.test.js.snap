// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> Should render and match the snapshot 1`] = `
<div
  class="field"
>
  <label
    class="styled__StyledLabel-oaisvy-0 fCsvBu"
  >
    Label
  </label>
  <div
    class="control"
  >
    <textarea
      class="styled__TextArea-sc-1u3hlpt-0 dCvwWm"
    />
  </div>
</div>
`;
